src/java/org/apache/cassandra/db/lifecycle/Tracker.java

```diff
@@ -338,7 +338,27 @@ public Throwable dropSSTables(final Predicate<SSTableReader> remove, OperationTy
             }
             catch (Throwable t)
             {
-                accumulate = abortObsoletion(obsoletions, accumulate);
+                logger.error("Failed to commit transaction for obsoleting sstables of {}", metadata.name, t);
+                Throwable err = abortObsoletion(obsoletions, null);
+                if (err == null && cfstore != null && cfstore.isValid())
```

Comment:
In general I think we should avoid refactoring code while we're trying to catch up with `master`. When we have test failures I often look at the DSE and CC code side-by-side to figure out if there was an unintentional change made that broke the test, and that's much easier if we don't refactor as we port. Additionally, it introduces more chances for conflicts (probably minor conflicts, but still).

So on balance, I'd prefer to avoid refactoring at this stage of the CC project. When we start writing new code then we should refactor as we go.